Makefile

```diff
@@ -112,3 +120,25 @@ verify: check format test
 
 .PHONY: verify-extended
 verify-extended: install-requirements check-generate check format test-cov test-cov-clean
+
+#####################################################################
+# Rules for cnudie component descriptors dev setup #
+#####################################################################
+
+.PHONY: cnudie-docker-images
+cnudie-docker-images: TAGS="-t $(CNUDIE_IMAGE_REGISTRY)/$(NAME):$(EFFECTIVE_VERSION)"
+cnudie-docker-images: docker-build
+
+.PHONY: cnudie-docker-push
+cnudie-docker-push:
+	@echo "Pushing docker images for version $(EFFECTIVE_VERSION) to registry $(CNUDIE_IMAGE_REGISTRY)"
+	@if ! docker images $(CNUDIE_IMAGE_REGISTRY)/$(NAME) | awk '{ print $$2 }' | grep -q -F $(EFFECTIVE_VERSION); then echo "$(CNUDIE_IMAGE_REGISTRY)/$(NAME) version $(EFFECTIVE_VERSION) is not yet built. Please run 'make cnudie-docker-images'"; false; fi
+	@docker push $(CNUDIE_IMAGE_REGISTRY)/$(NAME):$(EFFECTIVE_VERSION)
+
+.PHONY: cnudie-cd-build-push
+cnudie-cd-build-push:
+	@EFFECTIVE_VERSION=$(EFFECTIVE_VERSION) ./hack/generate-cd.sh
+
+.PHONY: cnudie-create-installation
+cnudie-create-installation:
+	@EFFECTIVE_VERSION=$(EFFECTIVE_VERSION) ./hack/create-installation.sh
```

Comment:
Please add the missing newline at the end of the file.

Reply:
newline added